hw/syn/tools/dc/run-syn.tcl

```diff
@@ -58,6 +47,21 @@ exec mkdir -p ${REPDIR} ${DDCDIR} ${VLOGDIR} ${WORKLIB} ${RESULTDIR}
 # define work lib path
 define_design_lib WORK -path $WORKLIB
 
+########################
+## Library Setup      ##
+########################
+
+# if the foundry root is specified, some more library setup is needed.
+if {$FOUNDRY_ROOT != ""} {
+  source "${FOUNDRY_ROOT}/syn/dc/setup.tcl"
+  # this PRIM_DEFAULT_IMPL selects the appropriate technology by defining
+  # PRIM_DEFAULT_IMPL=prim_pkg::Impl<tech identifier>
+  # PRIM_DEFAULT_IMPL is set inside the library setup script
+  set DEFINE "PRIM_DEFAULT_IMPL=${PRIM_DEFAULT_IMPL}+${PRIM_STD_CELL_VARIANT}"
+} else {
+  set DEFINE ""
```

Comment:
so for cdc, do we have to manually set `FOUNDRY_ROOT` to empty? if we don't define it, it looks like above will error?

Reply:
Yes, CDC will have an open-source view and a closed source one (like AscentLint), and when running the open-source one, the idea is to explicitly set the `FOUNDRY_ROOT` to an empty path `""` in the Dvsim Hjson config file.